zerver/views/__init__.py

```diff
@@ -253,27 +254,22 @@ def accounts_register(request):
 @zulip_login_required
 def accounts_accept_terms(request):
     # type: (HttpRequest) -> HttpResponse
-    email = request.user.email
-    domain = resolve_email_to_domain(email)
     if request.method == "POST":
         form = ToSForm(request.POST)
         if form.is_valid():
-            full_name = form.cleaned_data['full_name']
-            send_mail('Terms acceptance for ' + full_name,
-                    loader.render_to_string('zerver/tos_accept_body.txt',
-                        {'name': full_name,
-                         'email': email,
-                         'ip': request.META['REMOTE_ADDR'],
-                         'browser': request.META.get('HTTP_USER_AGENT', "Unspecified")}),
-                        settings.EMAIL_HOST_USER,
-                        ["all@zulip.com"])
-            do_change_full_name(request.user, full_name)
+            do_change_tos_version(request.user, settings.TOS_VERSION)
             return redirect(home)
-
     else:
         form = ToSForm()
+
+    email = request.user.email
+    domain = resolve_email_to_domain(email)
+    special_message_template = None
+    if request.user.tos_version is None and hasattr(settings, 'FIRST_TIME_TOS_TEMPLATE'):
```

Comment:
This `hasattr` line is incorrect.  You want `settings.FIRST_TIME_TOS_TEMPLATE is not None`, since we're set a default value of None for this settings in `zproject/settings.py`.  

A design goal of our settings system is to never need to do a `hasattr` check for a setting -- we always set a default, so you're always guaranteed you can access it (and check for None if appropriate).